src/InputToolbar.tsx

```diff
@@ -47,128 +47,96 @@ export interface InputToolbarProps<TMessage extends IMessage> {
   onPressActionButton?(): void
 }
 
-export default class InputToolbar<
-  TMessage extends IMessage = IMessage
-> extends React.Component<InputToolbarProps<TMessage>, { position: string }> {
-  static defaultProps = {
-    renderAccessory: null,
-    renderActions: null,
-    renderSend: null,
-    renderComposer: null,
-    containerStyle: {},
-    primaryStyle: {},
-    accessoryStyle: {},
-    onPressActionButton: () => {},
-  }
-
-  static propTypes = {
-    renderAccessory: PropTypes.func,
-    renderActions: PropTypes.func,
-    renderSend: PropTypes.func,
-    renderComposer: PropTypes.func,
-    onPressActionButton: PropTypes.func,
-    containerStyle: StylePropType,
-    primaryStyle: StylePropType,
-    accessoryStyle: StylePropType,
-  }
-
-  state = {
-    position: 'absolute',
-  }
-
-  keyboardWillShowListener?: EmitterSubscription = undefined
-  keyboardWillHideListener?: EmitterSubscription = undefined
-
-  componentDidMount() {
-    this.keyboardWillShowListener = Keyboard.addListener(
-      'keyboardWillShow',
-      this.keyboardWillShow,
-    )
-    this.keyboardWillHideListener = Keyboard.addListener(
-      'keyboardWillHide',
-      this.keyboardWillHide,
-    )
-  }
-
-  componentWillUnmount() {
-    if (this.keyboardWillShowListener) {
-      this.keyboardWillShowListener.remove()
+export function InputToolbar<TMessage extends IMessage = IMessage>(props: InputToolbarProps<TMessage>) {
+  const [position, setPosition] = useState('absolute');
+
+  useEffect(() => {
+    const keyboardWillShowListener = Keyboard.addListener('keyboardWillShow', () => setPosition('relative'));
+    return () => keyboardWillShowListener?.remove();
+  }, []);
+
+  useEffect(() => {
+    const keyboardWillHideListener = Keyboard.addListener('keyboardWillHide', () => setPosition('absolute'));
+    return () => keyboardWillHideListener?.remove();
+  }, []);
+
+  const renderActions = useCallbackOne(() => {
```

Comment:
I don't really like a callback returning a component, can you move it directly under the return, please?